src/main/python/resources.py

```diff
@@ -305,40 +322,32 @@ class Error:
 
         with Api(self.cxt, endpoint) as api:
             status, res = api.post(dat)
-
             if not res:
                 self.submit_hint.setText(Error.CONNECT_ERR)
                 return False
-
             if status == 200:
                 return True
-
-            if status == 500:
-                msg = ""
-                if res and "error" in res and "errmsg" in res["error"]:
-                    msg = res["error"]["errmsg"]
-                    if "E11000" in msg:
-                        msg = Error.E11000_ERR
-                else:
-                    msg = Error.UNKOWN_ERR
-
+            if (status == 400) or (status == 500):
+                msg = ''
+                for err in res.get('errors', []):
+                    error = Error.E11000_ERR if ('E11000' in err.get('errmsg', '')) else Error.UNKOWN_ERR
+                    msg += error + " "
                 self.submit_hint.setText(msg)
                 return False
+            else:
+                self.submit_hint.setText(Error.UNKOWN_ERR)
+                return False
+
 
     def _planning_check(self):
         # clean up hint
         self.reset_hint()
 
         # have to call function individually in order to raise hint
-        if not self._machine_name_check():
-            return False
-        if not self._cpu_gpu_check():
-            return False
-        if not self._cores_check():
-            return False
-        if not self._ram_check():
-            return False
-
+        if not self._machine_name_check(): return False
```

Comment:
Minor nit: If we don't care about which check fails, this could be a bit more "pythonic" using `all`. (I also think it makes it a bit more readable)
```python
# Returns true if all checks pass; false otherwise.
return all([
    self._machine_name_check(),
    self._cpu_gpu_check(),
    self._cores_check(),
    self._ram_check(),
])
```

Reply:
We do care which check fails. In each `_xxx_check()` we would raise error if the input is incorrect. If user have multiple error, we want to error shows by order (i.e. `machine_name` error then `cpu_gpu` error then ...)

Reply:
For this one I'm currently leaving it as is, I could go either way but I'm leaning towards what Martin said.  Although we are going to have to decide between code readability and UX

Reply:
> We do care which check fails. In each `_xxx_check()` we would raise error if the input is incorrect. If user have multiple error, we want to error shows by order (i.e. `machine_name` error then `cpu_gpu` error then ...)

I was speaking more towards the fact that we return `False` regardless of which check fails, so the change would be to instead return `False` if _any_ of the checks fails. We would then have to ensure that each respective check function handles it's error/success UX, but that's already what's happening. Regardless, it's just an ergonomic change so not terribly important.

Reply:
@samgomena I just decided to run a quick test and it looks like all() will evaluate each function for True/False before returning, which prevents us from short-circuiting the checks and forces us to run through all of them.  I don't think this is necessarily the best approach because if the first check fails, we don't have to evaluate the rest of the checks before continuing forward, which is more efficient.  See code below:

```
>>> def a():
...     print("a")
...     return True
...
>>> def b():
...     print("b")
...     return False
...
>>> def c():
...     print("c")
...     return True
...
>>> all([a(), b(), c()])
a
b
c
False
```